src/core/main.c

```diff
@@ -2726,6 +2726,8 @@ int main(int argc, char *argv[]) {
         Manager *m = NULL;
         FDSet *fds = NULL;
 
+        assert(argc > 0 && !isempty(argv[0]));
```

Comment:
Why only an assert and not an unconditional runtime check?
Shouldn't the subsequent code either handle an argc value of 0 just fine or not be called under any circumstance, since the calling method is out of scope.

Reply:
that's what the assert does?

Reply:
`assert` can be compiled out if NDEBUG is defined so it depends on how `systemd` is compiled. My guess would be that @cgzones hinted at `assert_se` or something like that.